### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by nevisende**
# Here's an overview of this PR:
- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post. R
- Added authentication to your API endpoints.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -23,6 +25,28 @@ def destroy
...
+  def add_comment
+    comment = Comment.new(author: @curr_user, post_id: params[:post_id], text: params[:text])
+
+    respond_to do |format|
+      if comment.save
+        format.json { render json: comment }
+      else
+        format.json { render json: { success: false, message: comment.errors.full_messages } }
+      end
+    end
```
Comment (andersonlebon) [inline app/controllers/comments_controller.rb:45]: image

- [ ] Nice Job :clap: :clap: on providing all API-endpoints(registration, login,...), however  the endpoint for creating a new comment is throwing an error :point_up: . Kindly fix it.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @team

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (nevisende) [reply app/controllers/comments_controller.rb:45]: @andersonlebon Kindly fix it? man if we know how to fix it why do you review it. And If you cant solve please confirm our pr

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- You have done a great job working on the requested changes 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nevisende** [comment]
Thank you @elisha2kyakpo1.

**Pull request merged**